src/Core/Context/CurrentContext.cs

```diff
@@ -392,15 +396,34 @@ public async Task<bool> ManageResetPassword(Guid orgId)
                     && (o.Permissions?.ManageResetPassword ?? false)) ?? false);
     }
 
-    public async Task<bool> ManageBilling(Guid orgId)
+    public async Task<bool> ViewSubscription(Guid orgId)
     {
-        var orgManagedByProvider = await ProviderIdForOrg(orgId) != null;
+        var orgManagedByMspProvider = (await _providerRepository.GetByOrganizationIdAsync(orgId))?.Type == ProviderType.Msp;
```

Comment:
The benefit of the `GetProviderOrganizations` helper in this class is that it caches its results. 

I suggest adding `ProviderType` to `ProviderUserOrganizationDetails` so that you can use the existing helper method and avoid adding another repository dependency.

Reply:
I tried using that but it doesn't work if the user has no `ProviderUser` associated to that `Organization`. Here we just need to know if the `Organization` has a provider, it doesn't matter if the user has a `ProviderUser`.